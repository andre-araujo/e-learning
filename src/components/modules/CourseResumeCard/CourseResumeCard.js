import React from 'react';
import { string, bool, number } from 'prop-types';
import Link from 'next/link';
import moment from 'moment';

import {
  Container,
  Thumbnail,
  Content,
  Description,
  Joined,
} from './CourseResumeCard.styles';
import Text from '../../elements/Text';
import Rating from '../../features/Rating/Rating';

function CourseResumeCard({
  id,
  joined,
  name,
  instructorName,
  category,
  keyWords,
  updated_at,
  created_at,
  thumbnail,
  rating,
}) {
  const formatedCreatedAt = moment(created_at).format('DD/MM/YYYY');
  const formatedUpdatedAt = moment(updated_at).format('DD/MM/YYYY');

  return (
    <Link href={{
      pathname: '/courses',
      query: { id },
    }}
    >
      <Container>
        <Thumbnail src={thumbnail} />
        <Content>
          <Text.Subtitle tag="h1" margin="0 0 0.5em 0">
            {name}
          </Text.Subtitle>

          <Description>
            <Text.Paragraph margin="0 0 0.5em 0">
              {/* {name} */}
            </Text.Paragraph>
          </Description>

          <p>
            <Text.Small>
              Categoria:
              {' '}
              {category}
              {
                keyWords && ` | Palavras-chave: ${keyWords}`
              }
            </Text.Small>
          </p>

          <Text.Small margin="0 0 10px">
          Instrutor:
            {' '}
            {instructorName}
            {' '}
            -
            {' '}
            {formatedCreatedAt}
            {
              formatedCreatedAt !== formatedUpdatedAt
            && ` - última atualização: ${formatedUpdatedAt}`
            }
          </Text.Small>
          <Rating.Stars rate={3} rating={rating} />
        </Content>
        {
          joined && (
            <Joined>
              <Text.Small>
                Inscrito
              </Text.Small>
            </Joined>
          )
        }
      </Container>
    </Link>
  );
}

CourseResumeCard.defaultProps = {
  id: null,
  name: null,
  instructorName: null,
  category: null,
  keyWords: null,
  updated_at: null,
  created_at: null,
  thumbnail: null,
  joined: null,
  rating: null,
};

CourseResumeCard.propTypes = {
  id: string,
  name: string,
  instructorName: string,
  category: string,
  keyWords: string,
  updated_at: string,
  created_at: string,
  thumbnail: string,
  joined: bool,
  rating: number,
};

export default CourseResumeCard;
