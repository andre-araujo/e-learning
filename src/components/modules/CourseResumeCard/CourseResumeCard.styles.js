import styled, { css } from 'styled-components';

export const Container = styled.section`
  padding: 15px 0;
  border-top: 1px solid #eee;
  display: flex;
  position: relative;
  cursor: pointer;
  align-items: center;

  &:hover:after {
    content: '';
    position: absolute;
    height: 100%;
    width: 100vw;
    background: rgba(0, 0, 0, 0.03);
    top: 0;
    left: 50%;
    transform: translateX(-50%);
  }
`;

export const Thumbnail = styled.div`
  height: 120px;
  width: 200px;
  background: #666;

  ${({ src }) => css`
    background: url(${src}) cover;
  `}
`;

export const Content = styled.div`
  padding-left: 20px;
  position: relative;
  display: flex;
  flex-direction: column;
  flex: 1;
`;

export const Description = styled.strong`
  flex: 1;
`;

export const Joined = styled.div`
  /* flex: 1; */
`;
