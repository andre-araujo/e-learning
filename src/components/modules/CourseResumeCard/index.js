import CourseResumeCard from './CourseResumeCard';

export default CourseResumeCard;
