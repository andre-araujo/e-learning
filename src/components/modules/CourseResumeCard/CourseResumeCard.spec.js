import React from 'react';
import { shallow } from 'enzyme';

import CourseResumeCard from './CourseResumeCard';

describe('CourseResumeCard component', () => {
  it('should match snapshot', () => {
    expect(shallow(<CourseResumeCard />)).toMatchSnapshot();
  });

  it('should match snapshot with keywords', () => {
    expect(shallow(<CourseResumeCard keyWords="keyword" />)).toMatchSnapshot();
  });

  it('should match snapshot with formatedUpdatedAt', () => {
    expect(shallow(<CourseResumeCard created_at="2017-12-13" updated_at="2017-12-14" />)).toMatchSnapshot();
  });

  it('should match snapshot with joined', () => {
    expect(shallow(<CourseResumeCard joined />)).toMatchSnapshot();
  });
});
