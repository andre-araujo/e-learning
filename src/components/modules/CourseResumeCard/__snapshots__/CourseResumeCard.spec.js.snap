// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseResumeCard component should match snapshot 1`] = `
<Link
  href={
    Object {
      "pathname": "/courses",
      "query": Object {
        "id": null,
      },
    }
  }
>
  <styled.section>
    <styled.div
      src={null}
    />
    <styled.div>
      <Component
        margin="0 0 0.5em 0"
        tag="h1"
      />
      <styled.strong>
        <Component
          margin="0 0 0.5em 0"
        >
          sdadsaads
        </Component>
      </styled.strong>
      <p>
        <Component>
          Categoria:
           
        </Component>
      </p>
      <Component
        margin="0 0 10px"
      >
        Instrutor:
         
         
        -
         
        Invalid date
      </Component>
      <value
        rate={3}
        rating={null}
      />
    </styled.div>
  </styled.section>
</Link>
`;

exports[`CourseResumeCard component should match snapshot with formatedUpdatedAt 1`] = `
<Link
  href={
    Object {
      "pathname": "/courses",
      "query": Object {
        "id": null,
      },
    }
  }
>
  <styled.section>
    <styled.div
      src={null}
    />
    <styled.div>
      <Component
        margin="0 0 0.5em 0"
        tag="h1"
      />
      <styled.strong>
        <Component
          margin="0 0 0.5em 0"
        >
          sdadsaads
        </Component>
      </styled.strong>
      <p>
        <Component>
          Categoria:
           
        </Component>
      </p>
      <Component
        margin="0 0 10px"
      >
        Instrutor:
         
         
        -
         
        13/12/2017
         - última atualização: 14/12/2017
      </Component>
      <value
        rate={3}
        rating={null}
      />
    </styled.div>
  </styled.section>
</Link>
`;

exports[`CourseResumeCard component should match snapshot with joined 1`] = `
<Link
  href={
    Object {
      "pathname": "/courses",
      "query": Object {
        "id": null,
      },
    }
  }
>
  <styled.section>
    <styled.div
      src={null}
    />
    <styled.div>
      <Component
        margin="0 0 0.5em 0"
        tag="h1"
      />
      <styled.strong>
        <Component
          margin="0 0 0.5em 0"
        >
          sdadsaads
        </Component>
      </styled.strong>
      <p>
        <Component>
          Categoria:
           
        </Component>
      </p>
      <Component
        margin="0 0 10px"
      >
        Instrutor:
         
         
        -
         
        Invalid date
      </Component>
      <value
        rate={3}
        rating={null}
      />
    </styled.div>
    <styled.div>
      <Component>
        Inscrito
      </Component>
    </styled.div>
  </styled.section>
</Link>
`;

exports[`CourseResumeCard component should match snapshot with keywords 1`] = `
<Link
  href={
    Object {
      "pathname": "/courses",
      "query": Object {
        "id": null,
      },
    }
  }
>
  <styled.section>
    <styled.div
      src={null}
    />
    <styled.div>
      <Component
        margin="0 0 0.5em 0"
        tag="h1"
      />
      <styled.strong>
        <Component
          margin="0 0 0.5em 0"
        >
          sdadsaads
        </Component>
      </styled.strong>
      <p>
        <Component>
          Categoria:
           
           | Palavras-chave: keyword
        </Component>
      </p>
      <Component
        margin="0 0 10px"
      >
        Instrutor:
         
         
        -
         
        Invalid date
      </Component>
      <value
        rate={3}
        rating={null}
      />
    </styled.div>
  </styled.section>
</Link>
`;
